shopfloor/actions/packaging.py

```diff
@@ -51,3 +51,14 @@ def package_has_several_products(self, package):
 
     def package_has_several_lots(self, package):
         return len(package.quant_ids.lot_id) > 1
+
+    def is_complete_mix_pack(self, package):
+        """Check if a package is mixed and completely reserved.
+
+        Will return true if the package has multiple distinct products and
+        all the package quantities are reserved.
+        """
+        return (
+            self.package_has_several_products(package)
+            and package.move_line_ids == package.reserved_move_line_ids
```

Comment:
There is something wrong here. While testing, partially reserved packages where considered and show as complete mix pack

Reply:
I changed this supplementation and added some tests